src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ChunkWindowManager.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * ChunkWindowManager.java
+ *
+ * Copyright (C) 2009-15 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.views.source.editors.text;
+
+import org.rstudio.core.client.*;
+import org.rstudio.studio.client.common.satellite.SatelliteManager;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+
+@Singleton
+public class ChunkWindowManager
+{
+   @Inject
+   public ChunkWindowManager(
+         Provider<SatelliteManager> pSatelliteManager)
+   {
+      pSatelliteManager_ = pSatelliteManager;
+   }
+
+   public void openChunkWindow(String docId, String chunkId)
```

Comment:
We've got some existing code that can be used to maximize the amount of space taken (up to some maximum reasonable size). In general our philosophy with zoom is to take up as much screen space as possible -- i.e. it's an immersive view to see as much content as possible). Here's how we handle some other satellite window types:

https://github.com/rstudio/rstudio/blob/ebb054efe3e650de818a251b11087311e5169ed8/src/gwt/src/org/rstudio/studio/client/htmlpreview/HTMLPreview.java#L39

https://github.com/rstudio/rstudio/blob/ebb054efe3e650de818a251b11087311e5169ed8/src/gwt/src/org/rstudio/studio/client/workbench/views/vcs/git/GitPresenter.java#L204

Those are just using larger default sizes. We also have a class named "ZoomUtils" that will help calculate a "greedy" zoom size (preserving aspect ratio if you want) that respect the current size of the screen:

https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/common/zoom/ZoomUtils.java

`getZoomWindowSize` respect aspect ratio, `getZoomedSize` just enforces a minimum and maximum.